src/socket.c

```diff
@@ -22,112 +24,159 @@
    \return The new socket's file descriptor
 */
 static int bind_socket(const fastd_bind_address_t *addr) {
+	const int zero = 0;
+	const int one = 1;
 	int fd = -1;
 	int af = AF_UNSPEC;
+	fastd_peer_address_t bind_address = addr->addr;
 
-	if (addr->addr.sa.sa_family != AF_INET) {
+	const int v6only = bind_address.sa.sa_family == AF_INET6 || addr->sourceaddr.sa.sa_family == AF_INET6;
+	if (!fastd_peer_address_is_v4_multicast(&bind_address)) {
```

Comment:
What is the reason you change the logic to use `PF_INET6` for IPv4 by default? The `PF_INET` fallback must work anyways for systems without IPv6 support, so I think it's a good idea to do that by default when IPv6 is not needed, so the `PF_INET` code paths receive more testing.